scripts/evaluation_script.py

```diff
@@ -48,3 +53,71 @@
     pic_loc="output/",
     pic_name="weekly_predicted_uptake.png",
 )
+
+# evaluate time-varying MSPE #
+date_mspe = pl.DataFrame()
+
+for i in range(len(rt_pred_2023)):
+
+    mspe = get_mspe(
+        data_df = nis_usa_2023,
+        pred_df = rt_pred_2023[i],
+        var = 'daily'
+    )
+    date_mspe = pl.concat(
+        [date_mspe,mspe],
+        how = 'vertical'
+    )
+
+# plot MSPE #
+time_axis = alt.Axis(
+    format = '%Y-%m',tickCount='month'
+)
+alt.Chart(date_mspe).mark_circle(color='black').encode(
+    x=alt.X('date:T',title='Date',axis=time_axis),
+    y=alt.Y('mspe',title='MSPE')
+).save(
+    'output/MSPE.png'
+)
+
+# evaluate predicted end-of-season uptake #
+date_eos = pl.DataFrame()
+
+for i in range(len(rt_pred_2023)):
+    eos = get_eos(
+        rt_pred_2023[i],
+        var = 'cumulative'
+    ).with_columns(
+        date = rt_pred_2023[i].filter(
+            pl.col('date') == pl.col('date').min()
+        ).select('date')
```

Comment:
what's going on here? this is getting the start of the season date?

Reply:
This is the getting the start of the forecast date. So we will know how the predicted end-of-season uptake changes over time.

Reply:
oh, I see, so the `i` is an index over forecasting dates?

I think it would be easier to understand if the models produce data frames with columns like forecasting date, observation date, and forecasted estimate. Because right now the starting forecast date isn't explicit, it's just implicit in being the first row of the data frame of predictions